### ISSUE TITLE
Use Django permission for event permissions instead of ChoiceField

### ISSUE THREAD

**Issue by JobDoesburg** [svthalia/concrexit#1342]
### Describe the change
Currently, the `Profile` model contains a field that states whether `Member`s  are allowed to attend borrels and/or events. This information, however, is only used/relevant for the `events` app. Therefore it would be nicer to separate logic, just as we did with `PaymentUser`s in `payments`:

- create an `EventUser` proxy model of `Member`
- add custom Django permissions `can_attend_borrels` and `can_attend_non-borrel_events`
- create a custom `EventUserAdmin` with actions to manage this (like in `payments`)

Basically just redo #1320 / #1277 on `payments`

### Motivation
Separation of duties. The `Member` model shouldn't contain information that only has a meaning in `Events`

**Comment by se-bastiaan** [comment]
Personally I do not believe you really need a separate admin with actions for this, since there are only a few users that are affected by these changes.

**Comment by se-bastiaan** [comment]
I'm not sure if this is really the best idea for the `can_attend_borrels` value. Since we don't use it in the website it should either be removed or kept the way it is.

I have no clue if it is even used.

**Comment by JobDoesburg** [comment]
We are using a permission for `can use Thalia Pay` so I still think this could be nice...

Maybe we should refactor `can attend events` to `can place event registrations` as to explain the functionality a bit better.

**Comment by se-bastiaan** [comment]
No. That is a bad idea. Then you can never use the sales app for anything else. If you want to do that I suggest you start thinking about doing it in a way that does not increase the circular dependencies.

But yes, the permission idea is still better. But I am really against subclassing the Member object in another place.

**Comment by JobDoesburg** [comment]
> No. That is a bad idea. Then you can never use the sales app for anything else. If you want to do that I suggest you start thinking about doing it in a way that does not increase the circular dependencies.

Yes that's exactly the problem. I see the sarcasm went missing in my comment (also because I do think it's actually a problem, but not a problem that can be solved in a nice way)

**Comment by pingiun** [comment]
We can do this the same way as blacklisted thaliapay users

**Comment by JobDoesburg** [comment]
Use a event blacklist, similar to the Thalia Pay user blacklist

**Comment by se-bastiaan** [comment]
```
class BlockedUser(models.Model):
    member = models.OneToOneField(Member, on_delete=models.CASCADE,)
    categories = models.ManyToManyField(EventCategory, on_delete=models.SET_NULL,)
